scripts/index.js

```diff
@@ -245,56 +252,114 @@ function exportToWebsite(options) {
+    let close = () => { };
+    if (suite === 'visual') {
+        const { url, server } = await startGoldensServer();
+        close = () => server.close();
+        data.proxies['/goldens'].target = url;
     }
-    return new Promise((resolve, reject) => {
-        try {
-            var p = cp.spawn(args.join(' '), { cwd: wd, env: process.env, shell: true, stdio: 'pipe' });
-            let clearLines = 0;
-            process.stdout.write(ansiEscapes.cursorHide);
-            p.stdout.on('data', function (data) {
-                data = _.compact(data.toString().trim().split(/\n/));
-                data.forEach(line => {
-                    if (clearLines > 0 && !options.verbose) {
-                        process.stdout.write(ansiEscapes.cursorUp(1));
-                        process.stdout.write(ansiEscapes.eraseDown);
-                    }
-                    if (line.startsWith('ok')) {
-                        clearLines = 1;
-                        console.log(chalk.green(line));
-                    }
-                    else if (line.startsWith('not ok')) {
-                        clearLines = 0;
-                        console.log(chalk.bold(chalk.red('not ok') + line.slice(6)));
-                    }
-                    else {
-                        clearLines = 0;
-                        console.log(line);
-                    }
-                });
-            });
-            p.stdout.once('end', () => {
-                process.stdout.write(ansiEscapes.cursorDown());
-                process.stdout.write(ansiEscapes.cursorShow);
-                resolve();
-            });
-        } catch (error) {
-            process.stdout.write(ansiEscapes.cursorDown());
-            process.stdout.write(ansiEscapes.cursorShow);
-            reject(error);
-        }
-    });
+    fs.writeFileSync(tempConfig, JSON.stringify({
+        ...data,
+        qunit: options,
+        tap_failed_tests_only: !options.verbose
+    }, null, '\t'));
+    // args
+    const port = options.port || suite === 'visual' ? 8081 : 8080;
+    try {
+        await kill(port);
+    } catch (error) {
+        
+    }
+    const args = ['testem', options.ci ? 'ci' : '', '--port', port, '-f', tempConfig, '-l', options.context.map(_.upperFirst)];
+    // env
+    // process.env.QUNIT_DEBUG_VISUAL_TESTS = options.debug;
+    // process.env.QUNIT_RECREATE_VISUAL_REFS = options.recreate;
+    // if (options.filter) {
+    //     process.env.QUNIT_FILTER = options.filter;
+    // }
+    // open localhost
+    const url = `http://localhost:${port}/`;
+    const start = (os.platform() == 'darwin' ? 'open' : os.platform() == 'win32' ? 'start' : 'xdg-open');
+    options.launch && cp.exec([start, url].join(' '));
+    // run
+    cp.spawn(args.join(' '), {
+        cwd: wd,
+        env: process.env,
+        shell: true,
+        stdio: 'pipe',
+        detached: !options.ci
```

Comment:
how do I close a detached process when the parent process dies?
It must be detached...